Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +15,219 @@ public static class BalanceStats
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+
+            Dictionary<string, List<float>> peopleBalances = ArrayOfPeople(peopleAndBalances);
+            List<string> names = new List<string>();
+            var biggestLastBalance = float.MinValue;
+
+            foreach (var personBalances in peopleBalances)
+            {
+                {
+                    if (personBalances.Value.Last() > biggestLastBalance)
+                    {
+                        names.Clear();
+                        biggestLastBalance = personBalances.Value.Last();
+                        names.Add(personBalances.Key);
+                    }
+                    else if (personBalances.Value.Last() == biggestLastBalance)
+                    {
+                        names.Add(personBalances.Key);
+                    }
+                }
+            }
+
+            string peopleString = BuildNamesString(names);
+            if (names.Count > 1)
+            {
+                return $"{peopleString} are the richest people. {BuildCurrancy(biggestLastBalance)}.";
+            }
+            return $"{peopleString} is the richest person. {BuildCurrancy(biggestLastBalance)}.";
         }
 
+
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+
+            Dictionary<string, List<float>> peopleBalances = ArrayOfPeople(peopleAndBalances);
+            List<string> names = new List<string>();
+            var lowestBalanceEver = float.MaxValue;
+
+            foreach (var personBalances in peopleBalances)
+            {
+                foreach (var balance in personBalances.Value)
+                {
+                    if (personBalances.Value.Last() < lowestBalanceEver)
+                    {
+                        names.Clear();
+                        lowestBalanceEver = balance;
+                        names.Add(personBalances.Key);
+                    }
+                    else if (personBalances.Value.Last() == lowestBalanceEver)
+                    {
+                        names.Add(personBalances.Key);
+                    }
+                }
+            }
+
+            if (names.Count > 1)
+            {
+                return $"{BuildNamesString(names)} have the least money. {BuildCurrancy(lowestBalanceEver)}.";
+            }
+
+            return $"{BuildNamesString(names)} has the least money. {BuildCurrancy(lowestBalanceEver)}.";
+        }
+
+
+        public static Dictionary<string, List<float>> ArrayOfPeople(string[] peopleAndBalances)
```

Comment:
ArrayOfPeopleWhat? What do you do with it? I think you are Reading or Creating People.